akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/AdaptedClusterImpl.scala

```diff
@@ -147,9 +150,17 @@ private[akka] final class AdapterClusterImpl(system: ActorSystem[_]) extends Clu
 
   // must not be lazy as it also updates the cached selfMember
   override val subscriptions: ActorRef[ClusterStateSubscription] =
-    system.internalSystemActorOf(subscriptionsBehavior(classicCluster), "clusterStateSubscriptions", Props.empty)
+    system.internalSystemActorOf(
+      // resume supervision: has state that shouldn't be lost in case of failure
+      Behaviors.supervise(subscriptionsBehavior(classicCluster)).onFailure(SupervisorStrategy.resume),
+      "clusterStateSubscriptions",
+      Props.empty)
 
   override lazy val manager: ActorRef[ClusterCommand] =
-    system.internalSystemActorOf(managerBehavior(classicCluster), "clusterCommandManager", Props.empty)
+    system.internalSystemActorOf(
+      // restart supervision: no state lost in case of failure
+      Behaviors.supervise(managerBehavior(classicCluster)).onFailure(SupervisorStrategy.restart),
+      "clusterCommandManager",
+      Props.empty)
```

Comment:
this is the real change, rest is just indentation

Reply:
Nice addition.